### PR TITLE
#5552 Check Filters on Sample page work as expected

### PR DESCRIPTION
**Pull request by VGitVlad**
allure
Closing  #5552

### CODE REVIEW

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/helpers/WebDriverHelpers.java [hunk 1]
```diff
@@ -332,6 +332,11 @@ public String getCheckedOptionFromHorizontalOptionGroup(By options) {
     return baseSteps.getDriver().findElement(options).findElement(SELECTED_RADIO_BUTTOn).getText();
   }
 
+  public String getCheckedOption(By options, By selected) {
```
Comment (BojteTamas) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/helpers/WebDriverHelpers.java:335]: `getCheckedOptionFromHorizontalOptionGroup` please use the implemented method

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/api/BusinessFlows.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    When(
+        "API: I create several new cases with a new sample foreach of them",
+        () -> {
+          for (int i = 0; i < number; i++) {
+            Person createPersonObject =
+                Person.builder()
+                    .uuid(UUID.randomUUID().toString())
+                    .firstName(faker.name().firstName())
+                    .lastName(faker.name().lastName())
+                    .build();
```
Comment (BojteTamas) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/api/BusinessFlows.java:59]: here you can make a person service, and we can reuse the code after

**Review by Mihai-Bota — APPROVED** [review]

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/helpers/WebDriverHelpers.java (hunk 1 above)
Comment (VGitVlad) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/helpers/WebDriverHelpers.java]: Done

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/api/BusinessFlows.java (hunk 2 above)
Comment (VGitVlad) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/api/BusinessFlows.java]: Done

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/LabUuid.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.Random;
+import lombok.Getter;
+
+@Getter
+public enum LabUuid {
+  VOREINGESTELLTES("VQL6NJ-HPJY24-56F2R5-T5UV2HUI"),
+  NEW_LAB("SORMAS-CONSTID-OTHERS-FACILITY");
+
+  private final String UuidValue;
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/LabUuid.java:29]: camelCase

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/NavBarPage.java [hunk 4]
```diff
@@ -28,4 +28,5 @@
   public static final By PERSONS_BUTTON = By.cssSelector("div#persons");
   public static final By DASHBOARD_BUTTON = By.cssSelector("div#dashboard");
   public static final By SAMPLE_BUTTON = By.cssSelector("[id='samples']");
+  public static final By CONFIRM_NAVIGATION = By.cssSelector(("[id=actionConfirm]"));
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/NavBarPage.java:31]: we can use By.id :D

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java [hunk 5]
```diff
@@ -37,7 +37,35 @@
   public static final By REASON_FOR_SAMPLING_TESTING_COMBOBOX =
       By.cssSelector("[id='samplingReason'] [class='v-filterselect-button']");
   public static final By FIELD_SAMPLE_ID_INPUT = By.cssSelector("[id='fieldSampleID']");
+  public static final By LABORATORY_COMBOBOX =
+      By.cssSelector("[id='lab'] [class='v-filterselect-button']");
+  public static final By LABORATORY_INPUT = By.cssSelector("[id='lab'] input");
+  public static final By LABORATORY_NAME_INPUT = By.cssSelector("[id='labDetails']");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java:43]: change with id

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java [hunk 6]
```diff
@@ -37,7 +37,35 @@
...
+  public static final By RECEIVED_OPTION_BUTTON = By.cssSelector("[id='received'] label");
+  public static final By DATE_SAMPLE_RECEIVED = By.cssSelector("[id='receivedDate'] input");
+  public static final By SPECIMEN_CONDITION_COMBOBOX =
+      By.cssSelector("[id='specimenCondition'] [class='v-filterselect-button']");
+  public static final By SPECIMEN_CONDITION_INPUT =
+      By.cssSelector("[id='specimenCondition'] input");
+  public static final By LAB_SAMPLE_ID_INPUT = By.cssSelector("[id='labSampleID']");
   public static final By COMMENT_AREA_INPUT = By.cssSelector("[id='comment']");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java:51]: change with id, both last 2 locators

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java [hunk 7]
```diff
@@ -37,7 +37,35 @@
...
+  public static final By TESTED_DISEASE_COMBOBOX =
+      By.cssSelector("[id='testedDisease'] [class='v-filterselect-button']");
+  public static final By TESTED_DISEASE_INPUT = By.cssSelector("[id='testedDisease'] input");
+  public static final By DATE_OF_RESULT = By.cssSelector("[id='testDateTime_date'] input");
+  public static final By TIME_OF_RESULT_INPUT = By.cssSelector("[id='testDateTime_time'] input");
+  public static final By TIME_OF_RESULT_COMBOBOX =
+      By.cssSelector("[id='testDateTime_time'] [class='v-filterselect-button']");
+  public static final By RESULT_VERIFIED_BY_LAB_SUPERVISOR_OPTIONS =
+      By.cssSelector(".popupContent #testResultVerified .v-select-option");
+  public static final By TEST_RESULTS_COMMENT_AREA_INPUT = By.cssSelector("[id='testResultText']");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java:68]: change with id

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/SampleManagementPage.java [hunk 8]
```diff
@@ -22,6 +22,22 @@
...
+      By.cssSelector("#samplePurpose .v-select-option");
+  public static final By TEST_RESULTS_SEARCH_COMBOBOX =
+      By.cssSelector("[id='pathogenTestResult'] [class='v-filterselect-button']");
+  public static final By LIST_OF_SAMPLES = By.cssSelector("[class='v-grid-tablewrapper'] tbody>tr");
+  public static final By SPECIMEN_CONDITION_SEARCH_COMBOBOX =
+      By.cssSelector("[id='specimenCondition'] [class='v-filterselect-button']");
+  public static final By LABORATORY_SEARCH_COMBOBOX =
+      By.cssSelector("[id='laboratory'] [class='v-filterselect-button']");
+  public static final By RESET_FILTER_BUTTON = By.cssSelector("[id='actionResetFilters']");
+  public static final By APPLY_FILTER_BUTTON = By.cssSelector("[id='actionApplyFilters']");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/SampleManagementPage.java:35]: change with id both 2 locators

**Review by rdutu-vg — APPROVED** [review]

**Pull request merged**